hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/api/ClusterResource.java

```diff
@@ -0,0 +1,64 @@
+package org.apache.mesos.hdfs.api;
+
+import org.apache.mesos.hdfs.config.HdfsFrameworkConfig;
+import org.apache.mesos.hdfs.state.LiveState;
+import org.apache.mesos.hdfs.state.AcquisitionPhase;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import com.google.inject.Inject;
+
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+/**
+ * Cluster management API.
+ */
+@Path("cluster")
+public class ClusterResource {
+  private final Log log = LogFactory.getLog(ClusterResource.class);
+
+  private HdfsFrameworkConfig config;
```

Comment:
If we're going to update the configuration while running, one solution is to update the mesos-site.xml file and copy the mesos-site.xml configuration file to the executors (similar to what we do for hdfs-site.xml). Otherwise we have an inconsistent state as you mentioned. In general, we need to clean up our configuration for this framework. This is a challenge since HDFS requires static configuration files.